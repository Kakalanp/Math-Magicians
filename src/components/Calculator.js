import { useState } from 'react';
import styles from './Calculator.module.css';
import calculate from '../logic/calculate';

export default function Calculator() {
  const [total, setTotal] = useState(0);
  const [next, setNext] = useState(null);
  const [operation, setOperation] = useState(null);

  function clickHandler(e) {
    const result = calculate({ total, next, operation }, e.target.textContent);
    setTotal(result.total);
    setNext(result.next);
    setOperation(result.operation);
  }
  const buttonList = ['AC', '+/-', '%', 'รท', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
  return (
    <div className={styles.page}>
      <div className={styles.text}>
        <p>Lets do some math!</p>
      </div>
      <div className={styles.calculator}>
        <div className={styles.calculator__result}>
          <p>
            {total}
            {' '}
            {operation}
            {' '}
            {next}
          </p>
        </div>
        {buttonList.map((item, index) => {
          let name = styles.calculator__button__grey;
          if ((index + 1) % 4 === 0 || index === 18) { name = styles.calculator__button__orange; }
          if (index === 16) { name = `${styles.calculator__button__grey} ${styles.calculator__button__zero}`; }
          return (
            <div key={item.id} className={name} onClick={clickHandler} onKeyPress={(e) => { if (e.key === 'Enter') clickHandler(); }} role="button" tabIndex={0}>
              <p style={{ userSelect: 'none' }}>{item}</p>
            </div>
          );
        })}
      </div>
    </div>
  );
}
