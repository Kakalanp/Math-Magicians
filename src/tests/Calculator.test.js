import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';
import Calculator from '../components/Calculator';

it('renders correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('displays the correct input on click', () => {
  const btn = render(<Calculator />).getByText('7');

  fireEvent.click(btn);

  expect(screen.getByTestId('display').textContent).toBe('  7');
});
