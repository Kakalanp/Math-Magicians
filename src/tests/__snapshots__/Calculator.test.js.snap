// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="page"
>
  <div
    className="text"
  >
    <p>
      Lets do some math!
    </p>
  </div>
  <div
    className="calculator"
  >
    <div
      className="calculator__result"
      data-testid="display"
    >
      <p>
        0
         
         
      </p>
    </div>
    <div
      className="calculator__button__grey"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <p
        style={
          Object {
            "userSelect": "none",
          }
        }
      >
        AC
      </p>
    </div>
    <div
      className="calculator__button__grey"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <p
        style={
          Object {
            "userSelect": "none",
          }
        }
      >
        +/-
      </p>
    </div>
    <div
      className="calculator__button__grey"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <p
        style={
          Object {
            "userSelect": "none",
          }
        }
      >
        %
      </p>
    </div>
    <div
      className="calculator__button__orange"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <p
        style={
          Object {
            "userSelect": "none",
          }
        }
      >
        รท
      </p>
    </div>
    <div
      className="calculator__button__grey"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <p
        style={
          Object {
            "userSelect": "none",
          }
        }
      >
        7
      </p>
    </div>
    <div
      className="calculator__button__grey"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <p
        style={
          Object {
            "userSelect": "none",
          }
        }
      >
        8
      </p>
    </div>
    <div
      className="calculator__button__grey"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <p
        style={
          Object {
            "userSelect": "none",
          }
        }
      >
        9
      </p>
    </div>
    <div
      className="calculator__button__orange"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <p
        style={
          Object {
            "userSelect": "none",
          }
        }
      >
        x
      </p>
    </div>
    <div
      className="calculator__button__grey"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <p
        style={
          Object {
            "userSelect": "none",
          }
        }
      >
        4
      </p>
    </div>
    <div
      className="calculator__button__grey"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <p
        style={
          Object {
            "userSelect": "none",
          }
        }
      >
        5
      </p>
    </div>
    <div
      className="calculator__button__grey"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <p
        style={
          Object {
            "userSelect": "none",
          }
        }
      >
        6
      </p>
    </div>
    <div
      className="calculator__button__orange"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <p
        style={
          Object {
            "userSelect": "none",
          }
        }
      >
        -
      </p>
    </div>
    <div
      className="calculator__button__grey"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <p
        style={
          Object {
            "userSelect": "none",
          }
        }
      >
        1
      </p>
    </div>
    <div
      className="calculator__button__grey"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <p
        style={
          Object {
            "userSelect": "none",
          }
        }
      >
        2
      </p>
    </div>
    <div
      className="calculator__button__grey"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <p
        style={
          Object {
            "userSelect": "none",
          }
        }
      >
        3
      </p>
    </div>
    <div
      className="calculator__button__orange"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <p
        style={
          Object {
            "userSelect": "none",
          }
        }
      >
        +
      </p>
    </div>
    <div
      className="calculator__button__grey calculator__button__zero"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <p
        style={
          Object {
            "userSelect": "none",
          }
        }
      >
        0
      </p>
    </div>
    <div
      className="calculator__button__grey"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <p
        style={
          Object {
            "userSelect": "none",
          }
        }
      >
        .
      </p>
    </div>
    <div
      className="calculator__button__orange"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <p
        style={
          Object {
            "userSelect": "none",
          }
        }
      >
        =
      </p>
    </div>
  </div>
</div>
`;
